app/src/main/java/com/duckduckgo/app/browser/state/BrowserApplicationStateInfo.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.state
+
+import android.app.Activity
+import android.os.Bundle
+import com.duckduckgo.app.global.ActivityLifecycleCallbacks
+import com.duckduckgo.browser.api.BrowserLifecycleObserver
+import com.duckduckgo.di.scopes.AppObjectGraph
+import com.squareup.anvil.annotations.ContributesMultibinding
+import java.util.*
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@ContributesMultibinding(AppObjectGraph::class)
+@Singleton
+class BrowserApplicationStateInfo @Inject constructor(
+    private val observers: Set<@JvmSuppressWildcards BrowserLifecycleObserver>
+) : ActivityLifecycleCallbacks {
+    private var created = 0
+    private var started = 0
+    private var resumed = 0
+
+    private val freshLaunchQueue: Queue<Boolean> = LinkedList()
```

Comment:
why is this a queue of `Boolean` vs a `Boolean`?

Reply:
no particular reason, in the beginning I though I would need to queue more events and then forgot to make it `Boolean?`. Will change